src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts

```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
   asset?.name?.toLowerCase()?.startsWith(filter?.toLowerCase()) ||
   asset?.symbol?.toLowerCase()?.startsWith(filter?.toLowerCase());
 
 export function useSearchCurrencyLists() {
   const { internalSelectedInputAsset: assetToSell, selectedOutputChainId } = useSwapContext();
 
-  const searchQuery = userAssetsStore(state => state.searchQuery);
+  const query = useSwapsStore(state => state.outputSearchQuery.trim().toLowerCase());
 
-  const [inputChainId, setInputChainId] = useState(assetToSell.value?.chainId ?? ChainId.mainnet);
-  const [toChainId, setToChainId] = useState(selectedOutputChainId.value);
-  const [assetToSellAddress, setAssetToSellAddress] = useState(
-    assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress']
-  );
+  const [state, setState] = useState({
+    assetToSellAddress: assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress'],
+    fromChainId: assetToSell.value?.chainId ?? undefined,
+    isCrosschainSearch: assetToSell.value ? assetToSell.value.chainId !== selectedOutputChainId.value : false,
+    toChainId: selectedOutputChainId.value ?? ChainId.mainnet,
+  });
 
-  const query = useMemo(() => searchQuery.trim().toLowerCase(), [searchQuery]);
-  const enableUnverifiedSearch = useMemo(() => searchQuery.length > 2, [searchQuery]);
+  const debouncedStateSet = useDebouncedCallback(setState, 500, { leading: true, trailing: true });
 
   useAnimatedReaction(
-    () => assetToSell.value,
+    () => ({
+      isCrosschainSearch: assetToSell.value ? assetToSell.value.chainId !== selectedOutputChainId.value : false,
+      toChainId: selectedOutputChainId.value ?? ChainId.mainnet,
+    }),
     (current, previous) => {
-      if (previous !== current) {
-        runOnJS(setInputChainId)(current?.chainId ?? ChainId.mainnet);
-        runOnJS(setAssetToSellAddress)(current?.[current?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress']);
+      if (previous && (current.isCrosschainSearch !== previous.isCrosschainSearch || current.toChainId !== previous.toChainId)) {
+        runOnJS(debouncedStateSet)({
+          assetToSellAddress: assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress'],
+          fromChainId: assetToSell.value?.chainId ?? undefined,
+          isCrosschainSearch: current.isCrosschainSearch,
+          toChainId: current.toChainId,
+        });
       }
     }
   );
 
-  useAnimatedReaction(
-    () => selectedOutputChainId.value,
-    (current, previous) => {
-      if (previous !== current) {
-        runOnJS(setToChainId)(current);
-      }
-    }
+  const sliceTopResultsByNetworkWithCoinIconUrls = useCallback(
+    (data: TokenSearchResult) => {
+      const matchingNetwork = data.filter(asset => asset.chainId === state.toChainId);
+      return matchingNetwork.slice(0, 20).filter(asset => asset.icon_url);
+    },
+    [state.toChainId]
```

Comment:
This is going to remove the other networks section as far as I can tell, haven't tested.